File: crates/facet_generate/src/tests/test_opaque_proxy_field/output.kt
package com.example

/// FFI view. Field holds a lattice type. Serialized as `UserId`.
data class SignInView(
    val userId: com.example.UserId,
)

/// Domain ID with `Facet`. Wire anchor for typegen.
data class UserId(
    val value: String,
)

/// Newtype wrapper around an opaque lattice field. Serialized as `UserId`.
data class UserIdWrapper(
    val value: com.example.UserId,
)

File: crates/facet_generate/src/tests/test_opaque_proxy_field/output.cs
using CommunityToolkit.Mvvm.ComponentModel;
using Facet.Runtime.Serde;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Example;

/// FFI view. Field holds a lattice type. Serialized as `UserId`.
public partial class SignInView : ObservableObject {
    [ObservableProperty]
    private UserId _userId;
}

/// Domain ID with `Facet`. Wire anchor for typegen.
public partial class UserId : ObservableObject {
    [ObservableProperty]
    private string _value;
}

/// Newtype wrapper around an opaque lattice field. Serialized as `UserId`.
public partial class UserIdWrapper : ObservableObject {
    [ObservableProperty]
    private UserId _value;
}
